From 7a769172a1f62e5198584dff9fd924e8a9e431a0 Mon Sep 17 00:00:00 2001
From: Cameron Gutman <aicommander@gmail.com>
Date: Wed, 2 Apr 2025 21:14:22 -0500
Subject: [PATCH] Fix null window on Qt 6.9

---
 app/gui/StreamSegue.qml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/app/gui/StreamSegue.qml b/app/gui/StreamSegue.qml
index 52cd9887e..29a8b2bce 100644
--- a/app/gui/StreamSegue.qml
+++ b/app/gui/StreamSegue.qml
@@ -176,7 +176,7 @@ Item {
             gc()
 
             // Run the streaming session to completion
-            session.exec(Window.window)
+            session.exec(window)
         }
 
         sourceComponent: Item {}
